src/boot/bootctl-uki.c

```diff
@@ -109,3 +111,74 @@ int verb_kernel_identify(int argc, char *argv[], void *userdata) {
         puts("unknown");
         return EXIT_SUCCESS;
 }
+
+static int read_pe_section(FILE *uki, struct PeSectionHeader *section,
+                           void **ret, int *ret_n) {
+        _cleanup_free_ void *data = NULL;
+        uint64_t soff;
+        uint32_t size;
+        uint32_t bytes;
+        int rc;
+
+        soff = le32toh(section->PointerToRawData);
+        size = le32toh(section->VirtualSize);
+
+        rc = fseek(uki, soff, SEEK_SET);
+        if (rc < 0)
+                return log_error_errno(errno, "seek to PE section");
+
+        data = malloc(size);
```

Comment:
oom check

Reply:
hmm, pretty much everywhere else we implicitly add an implicit unaccounted NUL byte to the end of such buffers, just in case someone by accident assumes it was a C string. maybe do the same here, i.e. just do:

```c
data = malloc(size + 1);
if (!data)
        return log_oom();

((uint8_t*) data)[size] = 0; /* safety NUL byte */
```